src/main/java/frc/robot/pioneersLib/bumSwerve/Encoder/EncoderIOSim.java

```diff
@@ -0,0 +1,50 @@
+package frc.robot.pioneersLib.bumSwerve.Encoder;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+
+public class EncoderIOSim implements EncoderIO {
+
+    int ID;
+    boolean inverted;
+    double encoderOffset;
+
+    EncoderSim encoder;
```

Comment:
delete encoder sim and just have a Rotation2d that's like "Rotation2d encoderValue" and update it in SwerveModule.java once u rebase to getSwerveModuleState.getRotation or whatever 👍🏽